flake.nix

```
@@ -0,0 +1,109 @@
+{
+  description = "Searx plugin for Searching the Green Web";
+
+  inputs.nixpkgs = {
+    type = "github";
+    owner = "NixOS";
+    repo = "nixpkgs";
+    ref = "21.05";
+  };
+
+  outputs = { self, nixpkgs }:
+    let
+      # System types to support.
+      supportedSystems = [ "x86_64-linux" ];
+      # Helper function to generate an attrset '{ x86_64-linux = f "x86_64-linux"; ... }'.
+      forAllSystems = f:
+        nixpkgs.lib.genAttrs supportedSystems (system: f system);
+      # Nixpkgs instantiated for supported system types.
+      nixpkgsFor = forAllSystems (system:
+        import nixpkgs {
+          inherit system;
+          overlays = [ self.overlay ];
+        });
+      in {
+
+        overlay = final: prev: rec {
+          tgwf-green-results-searx-plugin = prev.python38Packages.buildPythonPackage {
+            pname = "tgwf-searx-plugin";
+            version = "0.2";
+            buildInputs = [
+              prev.searx
+            ];
+            src = ./.;
+            doCheck = false;
+          };
+
+          # overlaying searx with version that depends plugin
+          # results in plugin being installed in the searx environment via python setuptools
+          # which is the requirement for the plugin to be available for searx:
+          # https://searx.github.io/searx/dev/plugins.html#external-plugins
+          searx = prev.searx.overrideAttrs (oldAttrs: {
+              propagatedBuildInputs = oldAttrs.propagatedBuildInputs ++ [
+                tgwf-green-results-searx-plugin
+              ];
+              postFixup = ''
+                ${oldAttrs.postFixup}
+                mkdir -p $out/share/static/plugins/external_plugins/plugin_only_show_green_results
+              '';
+            });
+
+        };
+
+        packages = forAllSystems (system:
+          {
+            inherit (nixpkgsFor.${system}) tgwf-green-results-searx-plugin searx;
+          }
+        );
+
+        # The default package for 'nix build'. This makes sense if the
+        # flake provides only one package or there is a clear "main"
+        # package.
+        # Returns Searx with plugin enabled
+        defaultPackage = forAllSystems (system: self.packages.${system}.searx);
+
+        # To use as `nix develop`
+        # will provide shell with
+        # - `searx-run`
+        # - simple searx config that would allow its launch & plugin utilization
+        devShell = forAllSystems (system: nixpkgsFor.${system}.mkShell rec {
+          packages = with nixpkgsFor.${system}; [ searx tgwf-green-results-searx-plugin ];
+          settingsFile = nixpkgsFor.${system}.writeText "settings.yml"
+            ''
+            use_default_settings: True
+            server:
+                secret_key : "dev-server-secret"
+
+            plugins:
+              - only_show_green_results
+            '';
```

Consider such indentation:
```suggestion
          settingsFile = nixpkgsFor.${system}.writeText "settings.yml" ''
            use_default_settings: True
            server:
                secret_key : "dev-server-secret"

            plugins:
              - only_show_green_results
          '';
```

Also, since this is a devshell maybe it would make more sense to have this file committed within repo and used from there?

---

I will in the future look at the possible pros and cons to separating this configuration part into separate file

One possible significant bonus from keeping it in the `flake.nix` is more ease with refacotirng,
since the configuration text references 
```
                plugins:
                  - ${pluginName}
```
in the refactored version, so that there aren't many hardcoded "only_show_green_results"